### PR TITLE
drivers: gpio: xec: Add SW debounce support for MEC15xx drivers

### PR DESCRIPTION
**Pull request by albertofloyd**
MEC15xx has no HW support for debouncing, add SW debounce algorithm to support GPIO_INT_DEBOUNCE flag.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

**Review by scottwcpg — APPROVED** [review]
I see two places where GPIO013 is touched is this left over debug code?

File: drivers/gpio/gpio_mchp_xec.c [hunk 1]
```diff
@@ -232,17 +285,23 @@ static void gpio_gpio_xec_port_isr(void *arg)
 	struct gpio_xec_data *data = dev->driver_data;
 	u32_t girq_result;
 	u32_t enabled_int;
+	u32_t enabled_debounce;
 
 	/* Figure out which interrupts have been triggered from the EC
 	 * aggregator result register
 	 */
+	GPIO_CTRL_REGS->CTRL_0013 ^= (1ul << 16);
```
Comment (franciscomunoz) [inline drivers/gpio/gpio_mchp_xec.c]: remove this line as mentioned by scott

File: drivers/gpio/gpio_mchp_xec.c [hunk 2]
```diff
@@ -329,6 +396,12 @@ static int gpio_xec_port040_076_init(struct device *dev)
 		gpio_gpio_xec_port_isr, DEVICE_GET(gpio_xec_port040_076), 0);
 
 	irq_enable(DT_GPIO_XEC_GPIO040_076_IRQ);
+
+	k_thread_create(&data->debounce_thread, data->debounce_thread_stack,
+			GPIO_TASK_STACK_SIZE,
+			debounce_thread_task, dev, NULL, NULL,
+			K_PRIO_COOP(4), 0, K_NO_WAIT);
```
Comment (franciscomunoz) [inline drivers/gpio/gpio_mchp_xec.c]: You don't need a k_thread_start if k_thread_create is using K_NO_WAIT

**Comment by albertofloyd** [comment]
Closing temporarily, since this solution may not be aligned with power management target.

**Pull request closed without merging**